src/SessionManager.ts

```diff
@@ -192,23 +191,25 @@ export class SessionManager {
             this.printedSessionState.printed = false
           }
 
-          // There is an internal error present in SAS Viya 3.5
-          // Retry to wait for a session status in such case of SAS internal error
           if (!sessionState) {
-            if (RETRY_COUNT < RETRY_LIMIT) {
-              RETRY_COUNT++
-
-              resolve(this.waitForSession(session, etag, accessToken))
-            } else {
-              reject(
-                new NoSessionStateError(
-                  responseStatus,
-                  this.serverUrl + stateLink.href,
-                  session.links.find((l: any) => l.rel === 'log')
-                    ?.href as string
-                )
+            const stateError = new NoSessionStateError(
+              responseStatus,
+              this.serverUrl + stateLink.href,
+              session.links.find((l: any) => l.rel === 'log')?.href as string
+            )
+
+            if (
```

Comment:
From here to line 207 doesn't seem to be serving any purpose, since `loggedErrors` is not used anywhere.

Reply:
we would like to log `NoSessionStateError` error with the same server response status code only one time

Reply:
In that case, since `loggedErrors` is outside of the `SessionManager` class and we never reset it, it will only be logged one time ever, not one time per call of `waitForSession`.

I think it should be an instance variable in `SessionManager` and we should reset it before returning from `waitForSession`.

Reply:
that makes sense, pushed changes